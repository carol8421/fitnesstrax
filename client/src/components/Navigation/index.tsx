import React from "react"
import { connect } from "react-redux"

import { classnames, ClassNames } from "../../classnames"
import * as redux from "../../redux"
import * as i18n from "../../i18n"
import { UserPreferences } from "../../settings"

interface Props {
  classes?: ClassNames
  prefs: UserPreferences
  view: redux.Views
  setView: (_: redux.Views) => void
}

const Navigation: React.SFC<Props> = ({
  classes,
  prefs,
  view,
  setView,
}: Props) => (
  <div className={classnames({ navigation: true, ...classes })}>
    <div>
      <a onClick={() => setView("History")}>
        {i18n.History.tr(prefs.language)}
      </a>
    </div>
    <div>
      <a onClick={() => setView("Preferences")}>
        {i18n.Preferences.tr(prefs.language)}
      </a>
    </div>
  </div>
)

export default Navigation
