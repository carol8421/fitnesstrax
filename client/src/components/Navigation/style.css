.navigation {
    display: flex;
}

.navigation > div {
    padding: 0.5em;
    cursor: pointer;
    margin: 1em;
}
