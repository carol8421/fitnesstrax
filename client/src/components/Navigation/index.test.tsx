import React from "react"
import { shallow } from "enzyme"

import Navigation from "./index"

describe("Navigation", () => {
  it("renders", () => {
    const setView = jest.fn()
    const wrapper = shallow(<Navigation view="History" setView={setView} />)
    console.log(wrapper.debug())
    expect(wrapper).toMatchSnapshot()
  })
})
