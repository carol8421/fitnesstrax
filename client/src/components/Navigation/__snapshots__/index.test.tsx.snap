// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders 1`] = `
<CenterPanel>
  <div
    className="navigation"
  >
    <div
      className="current"
    >
      <a
        onClick={[Function]}
      >
        History
      </a>
    </div>
    <div
      className=""
    >
      <a
        onClick={[Function]}
      >
        Preferences
      </a>
    </div>
  </div>
</CenterPanel>
`;
